sprouts.py

```diff
@@ -0,0 +1,246 @@
+                if DrawingLine:
+                    current_point = pygame.mouse.get_pos()
+                    if distance(start_line, current_point) > 30: #Only starts drawing if 30pix away from dot, resulting in less coordinates
+                        pygame.draw.line(win, ss.BLACK, start_line, current_point)
+                        start_line = current_point 
+                        current_line.append(start_line) #Saves line location
+                    
+                 
+        pygame.display.update()
+    
+        clock.tick(10) #10 Frames per second
+
+
+####FUNCTIONS
+
+#Writes all text 
+def text_writing(message,colour, x, y, size):
+    font = pygame.font.SysFont('Arial.ttf', size)
+    text = font.render(message, False, colour)
+    textRect = text.get_rect()
+    textRect.center = (x,y)
+    win.blit(text, textRect)
+
+#Updates text box surface - Will use to update score 
+def redraw_box():
+    pygame.draw.rect(win, ss.WHITE, (250, 5, 500, 46), 0)
+    pygame.display.update()
+
+#Displays error messages
+def error_display(error_line, error_dot):
+    if error_line:
+        redraw_box()
+        text_writing('Error: Too close to existing dot', ss.RED, 500, 25, 25)
+    
+    if error_dot:
+        redraw_box()
+        text_writing('Error: Start line on dot', ss.RED, 500, 25, 25)
+
+#Draws a dot anywhere, of any colour
+def draw_dot(position, dot_colour):
+    pygame.draw.circle(win, dot_colour, (position), 10, 10)
+    pygame.display.update()
+
+#Draws 3 random starter dots - Implements (faulty) distance check just in case of random overlap
+def start_sequence():
+    temp_list = []
+    for i in range(NumberStartingDots):
+        random_pos = (random.randint(100,900), random.randint(100,500)) #Manual playing field input to not overlap 'menu box', get too close to sides
+        temp_list.append(random_pos)
+
+    for pos in temp_list: #Checks only for exact same coordinates. Needs updating
+        if pos not in dot_pos:
+            dot_pos.append(pos)
+            draw_dot(pos, ss.RED)
+            print(dot_pos)
+        else:
+            print('Overlap')
+            start_sequence()
+
+    pygame.display.update()
+
+#Distance function
+def distance(dot1, dot2):
+    return math.sqrt((dot1[0] - dot2[0])**2 + (dot1[1] - dot2[1])**2)
+
+#Checks whether placed dot is too close to existing dot
+def check_dots(currentdotposition):
+    for dot in dot_pos:
+        if distance((dot), (currentdotposition)) < 30:
+            return False 
+    return True
+
+#Checks whether line starts on existing dot
+def check_line(start_line):
+    for dot in dot_pos:
+        if distance((dot), (start_line)) < 10:
+            return True
+    return False
+
+#These two check if a new dot is placed on a line. They don't yet work how I want them to. 
+#Calculates distance from point to line (line1 to line2). All arguments are "position" (x, y)
+def dist_point_to_line(line1, line2, point):
+    x0 = point[0]
```

Comment:
Maybe it makes sense to loop over the line coordinates (current_line). The current_line is a list with tuples with coordinates of the different positions the line can have. It looks like, now you are only checking the first two coordinates.
While in the loop you'll have to test for each line coordinate if it's within in certain threshold (you chose 80) of the position of the dot. The first time this test will say True, then the player made a 'good' click and you may create a new dot in the middle of the line, e.g. current_line[len(current_line)/2]. Could be useful to make a function for measuring the distance between two points (current_line[i] and dot_pos).
However, this is how I interpreted the game. Not sure whether it is correct :)